tofu/geom/_vignetting_tools.pyx

```diff
@@ -26,24 +26,33 @@ from . cimport _sorted_set as _ss
 cdef inline bint is_reflex(const double[3] u,
                            const double[3] v) nogil:
     """
+    /\
+     \
+     v\
+       \______>
+       P  u
     Determines if the angle between U and -V is reflex (angle > pi) or not.
+    We suppose the angle is defined anticlockwise from u to v
     Warning: note the MINUS in front of V, this was done as this is the only
              form how we will need this function. but it is NOT general.
+    Is reflex if
+            u x -v has a negative 3rd coordinate
+    https://dai.fmph.uniba.sk/upload/8/89/Gm17_lesson05.pdf
     """
     cdef int ii
     cdef double sumc
     cdef double[3] ucrossv
     # ...
-    _bgt.compute_cross_prod(u, v, ucrossv)
-    sumc = 0.0
-    for ii in range(3):
-        # normally it should be a sum, but it is a minus cause is we have (U,-V)
-        sumc = ucrossv[ii]
-    return sumc >= 0.
+    _bgt.compute_cross_prod(u, v, &ucrossv[0])
```

Comment:
why give a pointer (`&ucrossv[0]`) ? The routine takes an array (see definition)

Reply:
To be sure it will take the actual array and not a copy of it. So that the array is modified on exit. Not really necessary but some people prefer to do it to make sure everything is modified properly.
And an array is a pointer.